contracts/minter-suite/Minters/MinterSetPriceV5Merkle.sol

```diff
@@ -0,0 +1,684 @@
+                .checkDelegateForContract(
+                    msg.sender, // delegate
+                    _vault, // vault
+                    _coreContract // contract
+                );
+            require(isValidDelegee, "Invalid delegate-vault pairing");
+            vault = _vault;
+        }
+
+        // require valid Merkle proof
+        require(
+            MerkleLib.verifyAddress(
+                projectMerkleRoot[_coreContract][_projectId],
+                _proof,
+                vault
+            ),
+            "Invalid Merkle proof"
+        );
+
+        // limit mints per address by project
+        uint256 _maxProjectInvocationsPerAddress = _projectConfig
+            .useMaxInvocationsPerAddressOverride
+            ? _projectConfig.maxInvocationsPerAddressOverride
+            : MerkleLib.DEFAULT_MAX_INVOCATIONS_PER_ADDRESS;
+
+        // note that mint limits index off of the `vault` (when applicable)
+        require(
+            projectUserMintInvocations[_coreContract][_projectId][vault] <
+                _maxProjectInvocationsPerAddress ||
+                _maxProjectInvocationsPerAddress == 0,
+            "Maximum number of invocations per address reached"
+        );
+
+        // EFFECTS
+        // increment user's invocations for this project
+        unchecked {
+            // this will never overflow since user's invocations on a project
+            // are limited by the project's max invocations
+            projectUserMintInvocations[_coreContract][_projectId][vault]++;
+        }
+
+        tokenId = minterFilter.mint_joo(
+            _to,
+            _projectId,
+            _coreContract,
+            msg.sender
+        );
+
+        // invocation is token number plus one, and will never overflow due to
+        // limit of 1e6 invocations per project. block scope for gas efficiency
+        // (i.e. avoid an unnecessary var initialization to 0).
+        unchecked {
+            uint256 tokenInvocation = (tokenId % ONE_MILLION) + 1;
+            uint256 localMaxInvocations = _projectConfig.maxInvocations;
+            // handle the case where the token invocation == minter local max
+            // invocations occurred on a different minter, and we have a stale
+            // local maxHasBeenInvoked value returning a false negative.
+            // @dev this is a CHECK after EFFECTS, so security was considered
+            // in detail here.
+            require(
+                tokenInvocation <= localMaxInvocations,
+                "Maximum invocations reached"
+            );
+            // in typical case, update the local maxHasBeenInvoked value
+            // to true if the token invocation == minter local max invocations
+            // (enables gas efficient reverts after sellout)
+            if (tokenInvocation == localMaxInvocations) {
+                _projectConfig.maxHasBeenInvoked = true;
+            }
+        }
```

Comment:
I think this could be shoved into a `MaxInvocationsLib.sol` if we made one, since possible to do with only a `tokenID` and reference to the `ProjectConfig`